### PR TITLE
Databricks Volumes source connector

### PR DESCRIPTION
**Pull request by hubert-rutkowski85**
It adds the source connector and the test for it.

### CODE REVIEW

File: test_e2e/src/databricks-volumes.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+SCRIPT_DIR=$(dirname "$SRC_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=databricks-volumes
+OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$SCRIPT_DIR/workdir/$OUTPUT_FOLDER_NAME
+DOWNLOAD_DIR=$SCRIPT_DIR/download/$OUTPUT_FOLDER_NAME
+CI=${CI:-"false"}
+
+DATABRICKS_VOLUME="test-platform"
+DATABRICKS_VOLUME_PATH="databricks-volumes-test-output-24497"
```
Comment (potter-potter) [inline test_e2e/src/databricks-volumes.sh:15]: Lets find a better document to download. I think it ends up being a .json which I don't think ends up being Partitioned.

 I'll connect with Ronnie and put one up as soon as I can.

File: unstructured_ingest/v2/processes/connectors/databricks_volumes.py [hunk 2]
```diff
@@ -117,33 +115,137 @@ def path(self) -> str:
...
+class DatabricksVolumesIndexer(Indexer):
+    index_config: DatabricksVolumesIndexerConfig
+    connection_config: DatabricksVolumesConnectionConfig
     connector_type: str = CONNECTOR_TYPE
-    upload_config: DatabricksVolumesUploaderConfig
+
+    def precheck(self) -> None:
+        try:
+            self.connection_config.get_client()
+            logger.info("Indexer connection OK")
```
[the remark this answers is not in the document]
Comment (hubert-rutkowski85) [reply unstructured_ingest/v2/processes/connectors/databricks_volumes.py]: Yes, sorry, that was mine local testing helper, forgot about those. Removed.

File: test_e2e/src/databricks-volumes.sh (hunk 1 above)
Comment (hubert-rutkowski85) [reply test_e2e/src/databricks-volumes.sh:15]: Ok I modified the `test-databricks-volumes.py` script and with it I uploaded fake-memo to `/Volumes/utic-dev-tech-fixtures/default/test-platform/databricks-volumes-test-output-24497/fake-memo.pdf`

File: unstructured_ingest/v2/processes/connectors/databricks_volumes.py [hunk 3]
```diff
@@ -117,33 +115,137 @@ def path(self) -> str:
...
+            with self.connection_config.get_client().dbfs.download(path=file_data.identifier) as c:
+                read_content = c._read_handle.read()
+            with open(download_path, "wb") as f:
+                f.write(read_content)
+        except Exception as e:
+            logger.error(f"failed to download file {file_data.identifier}: {e}", exc_info=True)
+            raise SourceConnectionNetworkError(f"failed to download file {file_data.identifier}")
+
+        return DownloadResponse(
+            file_data=FileData(
```
[the remark this answers is not in the document]
Comment (hubert-rutkowski85) [reply unstructured_ingest/v2/processes/connectors/databricks_volumes.py]: Good point, changed.

**Pull request merged**